src/hooks/transactor/AddToBalanceTx.ts

```diff
@@ -0,0 +1,30 @@
+import { ProjectContext } from 'contexts/projectContext'
+import { UserContext } from 'contexts/userContext'
+import { BigNumber } from 'ethers'
+import { useContext } from 'react'
+
+import { TransactorInstance } from './Transactor'
+
+export function useAddToBalanceTx(): TransactorInstance<{
+  value: BigNumber
+}> {
+  const { transactor, contracts } = useContext(UserContext)
+  const { projectId } = useContext(ProjectContext)
+
+  return ({ value }, txOpts) => {
+    if (!transactor || !projectId || !contracts?.TicketBooth) {
+      if (txOpts?.onDone) txOpts.onDone()
```

Comment:
`txOpts?.onDone?.()` works just as well.

Reply:
updated. somehow after years of typescript i never knew that `fn.()` notation was a thing